public/app/features/dashboard-scene/settings/variables/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { QueryVariableEditor } from './editors/QueryVariableEditor';
+import { TextBoxVariableEditor } from './editors/TextBoxVariableEditor';
+
+interface EditableVariableConfig {
+  name: string;
+  description: string;
+  editor: React.ComponentType<any>;
+}
+
+export type EditableVariableType = Exclude<VariableType, 'system'>;
+
+export function isEditableVariableType(type: VariableType): type is EditableVariableType {
+  return type !== 'system';
+}
+
+const EDITABLE_VARIABLES: Record<EditableVariableType, EditableVariableConfig> = {
+  custom: {
+    name: 'Custom',
+    description: 'Define variable values manually',
+    editor: CustomVariableEditor,
+  },
+  query: {
+    name: 'Query',
+    description: 'Variable values are fetched from a datasource query',
+    editor: QueryVariableEditor,
+  },
+  constant: {
+    name: 'Constant',
+    description: 'Define a hidden constant variable, useful for metric prefixes in dashboards you want to share',
+    editor: ConstantVariableEditor,
+  },
+  interval: {
+    name: 'Interval',
+    description: 'Define a timespan interval (ex 1m, 1h, 1d)',
+    editor: IntervalVariableEditor,
+  },
+  datasource: {
+    name: 'Data source',
+    description: 'Enables you to dynamically switch the data source for multiple panels',
+    editor: DataSourceVariableEditor,
+  },
+  adhoc: {
+    name: 'Ad hoc filters',
+    description: 'Add key/value filters on the fly',
+    editor: AdHocFiltersVariableEditor,
+  },
+  textbox: {
+    name: 'Textbox',
+    description: 'Define a textbox variable, where users can enter any arbitrary string',
+    editor: TextBoxVariableEditor,
+  },
+};
+
+const EDITABLE_VARIABLES_SELECT_ORDER: EditableVariableType[] = [
+  'query',
+  'custom',
+  'textbox',
+  'constant',
+  'datasource',
+  'interval',
+  'adhoc',
+];
+
+export function getVariableTypeSelectOptions(): Array<SelectableValue<EditableVariableType>> {
+  return EDITABLE_VARIABLES_SELECT_ORDER.map((variableType) => ({
+    label: EDITABLE_VARIABLES[variableType].name,
+    value: variableType,
+    description: EDITABLE_VARIABLES[variableType].description,
+  }));
+}
+
+export function getVariableEditor(type: EditableVariableType) {
+  return EDITABLE_VARIABLES[type].editor;
+}
+
+interface CommonVariableProperties {
+  name?: string;
+  label?: string;
+}
+
+export function getVariableScene(type: EditableVariableType, initialState: CommonVariableProperties) {
+  switch (type) {
+    case 'custom':
+      return new CustomVariable(initialState);
+    case 'query':
+      return new QueryVariable(initialState);
+    case 'constant':
+      return new ConstantVariable(initialState);
+    case 'interval':
+      return new IntervalVariable(initialState);
+    case 'datasource':
+      return new DataSourceVariable(initialState);
+    case 'adhoc':
+      return new AdHocFilterSet(initialState);
+    case 'textbox':
+      return new TextBoxVariable({});
```

Comment:
```suggestion
      return new TextBoxVariable({initialState});
```

This was missing, probably that's the reason of the bug I found :)

Reply:
Ups, good catch! 👍 Those utils should have their tests 😄